pkg/scheduler/algorithm/priorities/balanced_resource_allocation.go

```diff
@@ -36,21 +37,36 @@ var (
 	BalancedResourceAllocationMap = balancedResourcePriority.PriorityMap
 )
 
-func balancedResourceScorer(requested, allocable *schedulercache.Resource) int64 {
+func balancedResourceScorer(requested, allocable *schedulercache.Resource, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
 	cpuFraction := fractionOfCapacity(requested.MilliCPU, allocable.MilliCPU)
 	memoryFraction := fractionOfCapacity(requested.Memory, allocable.Memory)
+	// This to find a node which has most balanced CPU, memory and volume usage.
+	if includeVolumes && utilfeature.DefaultFeatureGate.Enabled(features.VolumesOnNodeForBalancing) && allocatableVolumes > 0 {
+		volumeFraction := float64(requestedVolumes) / float64(allocatableVolumes)
+		if cpuFraction >= 1 || memoryFraction >= 1 || volumeFraction >= 1 {
+			// if requested >= capacity, the corresponding host should never be preferred.
+			return 0
+		}
+		// Compute variance for all the three fractions.
+		mean := (cpuFraction + memoryFraction + volumeFraction) / float64(3)
+		variance := float64(((cpuFraction - mean) * (cpuFraction - mean)) + ((memoryFraction - mean) * (memoryFraction - mean)) + ((volumeFraction - mean) * (volumeFraction - mean)))
```

Comment:
Here's the explanation of variance, not sure why we use it?

> In probability theory and statistics, variance is the expectation of the squared deviation of a random variable from its mean.

Reply:
So, we wanted to know how close CPU, memory utilization, no of volumes on nodes are to each other. The closer means the node is well balanced. I tried explaining the same here-https://github.com/kubernetes/kubernetes/issues/58232#issuecomment-364119456

I could have used standard deviation as well but I thought of stopping at variance.

Reply:
Great, got that.

Reply:
Maybe standard deviation is better; it'll also address Bobby's comments above :)

Reply:
Just deviding by 3 addresses my comment and since we want to compare one node against others, standard deviation and variance both work equally well here. Given that taking square root is computationally expensive, I would just use variance.

Reply:
Yes, computational cost was the reason that I stopped going towards standard deviation.

Reply:
ok to me :)